File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/entity/Team.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.streampark.console.system.entity;

import org.apache.streampark.console.base.mybatis.entity.BaseEntity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

import javax.validation.constraints.Size;

@Getter
@Setter
@EqualsAndHashCode(callSuper = true)
@TableName("t_team")
public class Team extends BaseEntity {

    @TableId(type = IdType.AUTO)
    private Long id;

    @Size(min = 4, max = 20, message = "{range}")
    private String teamName;

    private String description;

    private transient String createTimeFrom;

    private transient String createTimeTo;
}

File: streampark-console/streampark-console-webapp-v2/src/service/api/team.ts
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import type { TeamListRecord, TeamParam } from '@/types/api/system/model/teamModel'
import type { BasicTableParams } from '@/types/api/model/baseModel'
import { request } from '../http'

export function fetchTeamList(data: BasicTableParams & { teamName?: string }) {
    return request.Post<TeamListRecord[]>('/team/list', data)
}

export function fetchTeamCreate(data: TeamParam) {
    return request.Post('/team/post', data)
}

export function fetchTeamUpdate(data: TeamParam) {
    return request.Put('/team/update', data)
}

export function fetchTeamDelete(params: { id: string }) {
    const method = request.Delete<{ status: string }>('/team/delete')
    method.config.params = params
    return method
}
